dwave/cloud/computation.py

```diff
@@ -594,33 +674,79 @@ def occurrences(self):
             (-1, -1, 1, 1, ' --> ', -2.0, 28)
 
         """
-        self._load_result()
-        if 'occurrences' in self._result:
-            return self._result['occurrences']
+
+        # return num_occurrences from sampleset, if already constructed
+        result = self.result()
+        if 'sampleset' in result:
+            return result['sampleset'].record.num_occurrences
+
+        # fallback to num_occurrences from response
+        # (but `occurrences` data is not present if `answer_mode` was set to "raw")
+        if 'num_occurrences' in result:
+            return result['num_occurrences']
         elif self.return_matrix:
-            return np.ones((len(self._result['samples']),))
+            return np.ones((len(result['solutions']),))
         else:
-            return [1] * len(self._result['samples'])
+            return [1] * len(result['solutions'])
+
+    @property
+    def sampleset(self):
+        """Return :class:`~dimod.SampleSet` representation of the results."""
+
+        result = self._load_result()
+        if 'sampleset' in result:
+            return result['sampleset']
+
+        # construct sampleset from available data
+        try:
+            import dimod
+        except ImportError:
+            raise RuntimeError("Can't construct SampleSet without dimod. "
```

Comment:
Maybe keep this as an `ImportError` rather than  `RuntimeError`?

Reply:
Yes, that's a good point. I considered using `ImportError`, but re-throwing `ImportError` with a different message (having in mind this exception is used (almost?) exclusively by the `import` statement) seems wrong. Or at least not "obvious way to do it".

Due to exception nesting, user will be able to see the whole exception stack trace anyway, and see that the `RuntimeError` was caused by a failing import.

As a Python user I would be surprised to see `ImportError` raised because of another `ImportError` (like `imp` or `importlib` failed importing something...)

If you're concerned about handling this case programmatically, then subclassing (of `ImportError`) would probably be the best way. But I don't think we should overthink this because dimod might very well become dependency of the client in future.